Update figures

scharlottej13 opened the pull request:
A start at updating the docs with new figures, re https://github.com/dask/community/issues/135. Would love your input @jsignell and @jacobtomlinson!

GPUtester commented:
Can one of the admins verify this patch?

jacobtomlinson commented:
This is really exciting to see, thanks for working on this @scharlottej13.

One thing that came up in dask/community#135 was tweaking the colour contrast of the new figures for accessibility. Have the colours in this PR been changed to fix this?

scharlottej13 commented:
Thanks @jacobtomlinson! Yes, we're waiting on new figures with the improved color contrast, but once we have them I'll swap those in instead. Until then, happy to take feedback on figure placement (or anything else)

ian-r-rose commented on docs/source/graphs.rst:
```diff
@@ -42,6 +42,9 @@ execute this graph in a way that respects these data dependencies and leverages
 simultaneously.
 
+.. image:: images/map-reduce-task-scheduling.svg
+   :align: center
```
In the future, we'll be flagging these for alt text, for now we can deliberately defer :)

ian-r-rose replied:
Since Dask is represented by the "Full Task Scheduling" box in this image, perhaps we can name check that further down?

Perhaps something like

```
Dask is a specification that encodes full task scheduling with minimal incidental...
```
(fixes a typo as a bonus)

scharlottej13 replied:
done!

ian-r-rose commented on docs/source/index.rst:
```diff
@@ -29,7 +29,7 @@ Dask emphasizes the following virtues:
 
 
-.. image:: images/dask-overview.svg
+.. image:: images/dask-overview.png
```
I think we'll want these to be SVG before merging

scharlottej13 replied:
I asked Christiana about this image, she's going to reach out to the designer about getting the SVG version fixed so `Dask Array` is not in a different color. The rest look good, though!

ian-r-rose commented on docs/source/how-to/deploy-dask-clusters.rst:
```diff
@@ -6,8 +6,23 @@ locally on your own machine or on a distributed cluster.  If you are just
+
 Dask has two families of task schedulers:
 
+.. image:: ../images/dask-overview.png
```
I'd move this image below the scheduler list (as you did in the `index`), so that the types of schedulers immediately follows the above intro "Dask has two families of task schedulers"

jsignell replied:
Not sure what the stylistic guidelines are on this, but it might make sense to outline the schedulers section in the image.

jsignell replied:
```suggestion
.. image:: ../images/dask-overview.png
   :alt: Dask is composed of three parts. "Collections" create "Task Graphs" which are then sent to the "Scheduler" for execution. There are two types of schedulers that are described in more detail below. 
```

ian-r-rose commented on docs/source/how-to/deploy-dask-clusters.rst:
```diff
@@ -6,8 +6,23 @@ locally on your own machine or on a distributed cluster.  If you are just
+
+.. raw:: html
+
+   <iframe width="560"
```
Cool

ian-r-rose commented on docs/source/how-to/deploy-dask-clusters.rst:
```diff
@@ -34,18 +52,16 @@ scheduler you must set up a ``Client``
+the older single-machine scheduler.
 
-.. raw:: html
+.. figure:: ../images/dask-cluster-manager.svg
```
The figure uses phrases like "Resource Manager" and "Cluster Manager" which are not used in the text. I think it would be worth introducing those in a sentence or two above

scharlottej13 replied:
Added some explanatory text! Let me know what you think (and if it is correct). I also linked to some "Dask notes" blog posts-- not sure if we want to do this in the docs?

ian-r-rose reviewed (commented):
Thanks for this @scharlottej13! I had a few comments, but this is looking really nice!

Any thoughts about figure placement @jacobtomlinson or @jsignell?

jsignell commented on docs/source/how-to/adaptive.rst:
```diff
@@ -11,13 +11,24 @@ two situations:
+2.  The user may be very active, and is limited by their original allocation,
+    particularly in the case of bursty workloads
+
+.. image:: ../images/bursty-workloads.svg
```
If you are using this figure to represent just the notion of a bursty workload, then I think it would make more sense not to have the adaptive part in this figure.

jsignell replied:
also the sideways words on both of these are very small. Maybe they should get  larger width.

scharlottej13 replied:
good point. I removed the bursty workloads image from this page because I think the page really focuses on the adaptive part. Do you know of another place where the bursty workloads image might be a good fit in the docs?

jsignell replied:
I can't think of where it would go right now, but we can still include it in the images.

scharlottej13 replied:
ok! Removed here, but added to dask/marketing

jsignell commented on docs/source/how-to/deploy-dask-clusters.rst:
```diff
@@ -16,6 +31,9 @@ Dask has two families of task schedulers:
     more features, but also requires a bit more effort to set up.  It can
     run locally or distributed across a cluster.
 
+.. image:: ../images/distributed-overview.svg
```
This needs a sentence that makes it clear that distributed scheduler == distributed cluster.

scharlottej13 replied:
Thanks! I added this to the caption, let me know what you think!

scharlottej13 commented:
Thanks @ian-r-rose and @jsignell for the feedback! These updates include:
- images with the latest color palette (see https://github.com/dask/community/issues/135)
- addressing your comments
I'll also open up a PR adding the updated images to dask/marketing

scharlottej13 commented on docs/source/images/dask-overview_scheduler_border.svg:
```diff
@@ -0,0 +1,58 @@
+<svg width="1920" height="1080" viewBox="0 0 1920 1080" fill="none" xmlns="http://www.w3.org/2000/svg">
```
@jsignell this is for your border suggestion on [How to... Deploy Dask clusters](https://docs.dask.org/en/latest/how-to/deploy-dask-clusters.html). Maybe the box should go around the 'schedulers' text too?

jsignell replied:
Yeah I think around the schedulers too and maybe it needs a bit of text on the page that just calls out that we are talking about the schedulers section and that the distributed overview below is kind of like a zoomed in graphic about just the distributed scheduler. I am thinking about how they do in maps:

![image](https://user-images.githubusercontent.com/4806877/146049847-1d88ac26-c984-4ce1-995d-037af5de1473.png)

Not saying that we need to make something like that. I just think the graphics would be more easy to understand if we mention how they relate.

scharlottej13 replied:
Yup, that makes sense! I can adjust this a bit.

scharlottej13 commented:
Just opened up the related PR in marketing: https://github.com/dask/marketing/pull/12

scharlottej13 commented on docs/source/how-to/deploy-dask-clusters.rst:
```diff
@@ -16,7 +28,16 @@ Dask has two families of task schedulers:
-If you import Dask, set up a computation, and then call ``compute``, then you
+|
+
+.. figure:: ../images/dask-overview-distributed-callout.svg
```
@jsignell here's the new version of a combined Dask overview & Distributed overview images, based on your suggestion with the magnified map image

jsignell commented:
This is great! Thanks @scharlottej13!!
